CI: CodeQL scanning fails — autobuild uses JDK 17 but project requires JDK 21

## Summary

The default CodeQL workflow (`github/codeql-action`) fails during the autobuild step because it uses JDK 17, while this project requires JDK 21 (Android SDK 36, `sourceCompatibility = JavaVersion.VERSION_21`). The autobuild also lacks the `google-services.json` file and Gradle setup that the build needs.

## Symptoms

```
[ERROR] Spawned process exited abnormally (code 1; tried to run:
  [./gradlew, --no-daemon, ..., -Dorg.gradle.java.home=/usr/lib/jvm/temurin-17-jdk-amd64, clean])
```

```
We were unable to automatically build your code.
Please change the build mode for this language to manual and specify build steps.
```

The autobuild picks up the Gradle wrapper and attempts `./gradlew clean` with JDK 17. The build fails because:

1. **Wrong JDK version:** The project requires JDK 21 (`temurin-21`), but CodeQL's autobuild defaults to JDK 17.
2. **Missing `google-services.json`:** The build needs this file decoded from the `GOOGLE_SERVICES_JSON_BASE64` secret, just like `build-and-test.yml`.
3. **Missing Gradle setup:** The project uses `gradle/actions/setup-gradle@v5` with cache encryption for reproducible builds.

## Root Cause

No custom CodeQL workflow exists in `.github/workflows/`. GitHub's default CodeQL analysis uses `autobuild` mode for Java/Kotlin, which doesn't account for project-specific requirements (JDK version, secrets, Android SDK).

## Proposed Fix

Create a `.github/workflows/codeql.yml` workflow that uses `build-mode: manual` and mirrors the build setup from the existing `build-and-test.yml` workflow:

```yaml
name: "CodeQL"

on:
  push:
    branches: [ main, develop ]
  pull_request:
    branches: [ main, develop ]
  schedule:
    - cron: '0 6 * * 1'  # Weekly Monday 06:00 UTC

concurrency:
  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
  cancel-in-progress: ${{ github.event_name == 'pull_request' }}

jobs:
  analyze:
    name: Analyze (Java/Kotlin)
    runs-on: ubuntu-latest
    permissions:
      security-events: write
      contents: read
      actions: read
    env:
      OER_APP_ID_RELEASE: ${{ secrets.OER_APP_ID_RELEASE }}
      GRADLE_OPTS: >-
        -Dorg.gradle.jvmargs=-Xmx6g
        -Dorg.gradle.parallel=false
        -Dorg.gradle.daemon=false
        -Dorg.gradle.caching=true
    steps:
      - uses: actions/checkout@v6

      - name: Set up JDK 21
        uses: actions/setup-java@v5
        with:
          java-version: '21'
          distribution: 'temurin'

      - name: Decode google-services.json
        run: echo "${{ secrets.GOOGLE_SERVICES_JSON_BASE64 }}" | base64 --decode > ${{ github.workspace }}/app/google-services.json

      - name: Setup Gradle
        uses: gradle/actions/setup-gradle@v5
        with:
          cache-encryption-key: ${{ secrets.GRADLE_ENCRYPTION_KEY }}

      - name: Grant execute permission for gradlew
        run: chmod +x ./gradlew

      - name: Initialize CodeQL
        uses: github/codeql-action/init@v3
        with:
          languages: java-kotlin
          build-mode: manual

      - name: Build for CodeQL
        run: ./gradlew assembleDebug

      - name: Perform CodeQL Analysis
        uses: github/codeql-action/analyze@v3
        with:
          category: "/language:java-kotlin"
```

### Key decisions

| Concern | Decision |
|---|---|
| `build-mode` | `manual` — autobuild can't handle JDK 21 + Android + secrets |
| Build command | `assembleDebug` — enough to compile all sources for analysis without signing |
| JDK | `temurin-21` — matches `build-and-test.yml` |
| Schedule | Weekly Monday 06:00 UTC — catches new CVE patterns without excessive runs |
| Concurrency | Same pattern as `build-and-test.yml` — cancel in-progress for PRs only |

## Affected File

`.github/workflows/codeql.yml` (new file)

## References

- [CodeQL: Automatic build failed](https://docs.github.com/en/code-security/code-scanning/troubleshooting-code-scanning/automatic-build-failed)
- [CodeQL manual build mode](https://docs.github.com/en/code-security/code-scanning/creating-an-advanced-setup-for-code-scanning/codeql-code-scanning-for-compiled-languages#specifying-build-steps-manually)
- Existing build setup: `.github/workflows/build-and-test.yml`

--- a/.github/workflows/codeql.yml
+++ b/.github/workflows/codeql.yml
@@ -50,7 +50,7 @@ jobs:
           build-mode: manual
 
       - name: Build for CodeQL
-        run: ./gradlew assembleDebug
+        run: ./gradlew clean assembleDebug --no-build-cache --no-configuration-cache
 
       - name: Perform CodeQL Analysis
         uses: github/codeql-action/analyze@v3

